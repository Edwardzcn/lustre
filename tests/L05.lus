node test(x : real) returns (y : int; z : int)
let
  y = int x;
  z = floor x;
tel

