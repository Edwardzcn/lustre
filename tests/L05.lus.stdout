PhaseToCore
===========

node (x : real)
  returns (y, z)
  assumes ()
  shows ()
let
  y : int
    = IntCast(x)
   
  z : int
    = FloorCast(x)
tel

--- Step 1 ---
x : real  = 2.5
y = 2
z = 2
--- Step 2 ---
x : real  = 3.5
y = 3
z = 3
--- Step 3 ---
x : real  = -2.5
y = -2
z = -3
--- Step 4 ---
x : real  = -3.5
y = -3
z = -4
--- Step 5 ---
x : real  = (EOF)
